spec/enumerable_spec.rb

```diff
@@ -0,0 +1,85 @@
+# spec/enumerables_spec.rb
+
+require_relative '../my_methods.rb'
+
+RSpec.describe Enumerable do
+  let(:test1) { [5, 4, 2] }
+  let(:test2) { [3] }
+  let(:test3) { [nil, true, 10] }
+  let(:test4) { [1, 2, 3, 1, 2, 3, 4, 1, 2, 3] }
+
+  describe '#my_each' do
+    it 'should interate through an array' do
+      expect(test1.my_each { |x| p x }).to eql(test1.each { |x| p x })
+      expect(test1.my_each { |x| p x * 2 }).to eql(test1.each { |x| p x * 2 })
+      expect(test1.my_each { |x| p x * 4 }).to eql(test1.each { |x| p x * 4 })
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'should interate through an array and its index' do
+      expect(test1.my_each_with_index { |x, y| x + y }).to eql(test1.each_with_index { |x, y| x + y })
+      expect(test1.my_each_with_index { |x, y| x * y }).to eql(test1.each_with_index { |x, y| x * y })
+      expect(test1.my_each_with_index { |x, y| x - y }).to eql(test1.each_with_index { |x, y| x - y })
```

Comment:
You should have only one `it` block per `expected` behavior ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png)  from  Pathwright).